modules/nodes-scheduler-default-predicates.adoc

```diff
@@ -22,87 +22,102 @@ name.
 
 The default scheduler policy includes the following predicates:
 
-**_NoVolumeZoneConflict_** checks that the volumes a pod requests
+*NoVolumeZoneConflict* checks that the volumes a pod requests
 are available in the zone.
 ----
 {"name" : "NoVolumeZoneConflict"}
 ----
 
-**_MaxEBSVolumeCount_** checks the maximum number of volumes that can be attached to an AWS instance.
+*MaxEBSVolumeCount* checks the maximum number of volumes that can be attached to an AWS instance.
 ----
 {"name" : "MaxEBSVolumeCount"}
 ----
 
-**_MaxGCEPDVolumeCount_** checks the maximum number of Google Compute Engine (GCE) Persistent Disks (PD).
+*MaxAzureDiskVolumeCount*  checks the maximum number of Azure Disk Volumes.
 ----
-{"name" : "MaxGCEPDVolumeCount"}
+{"name" : "MaxAzureDiskVolumeCount"}
 ----
 
-**_MatchInterPodAffinity_** checks if the pod affinity/anti-affinity rules permit the pod.
+*PodToleratesNodeTaints* checks if a pod can tolerate the node taints.
 ----
-{"name" : "MatchInterPodAffinity"}
+{"name" : "PodToleratesNodeTaints"}
+----
+
+*CheckNodeUnschedulable* checks if a pod can be scheduled on a node with `Unschedulable` spec.
+----
+{"name" : "CheckNodeUnschedulable"}
+----
+
+
+*CheckVolumeBinding* evaluates if a pod can fit based on the volumes, it requests, for both bound and unbound PVCs.
+* For PVCs that are bound, the predicate checks that the corresponding PV's node affinity is satisfied by the given node.
+* For PVCs that are unbound, the predicate searched for available PVs that can satisfy the PVC requirements and that
+the PV node affinity is satisfied by the given node.
+
+The predicate returns true if all bound PVCs have compatible PVs with the node, and if all unbound PVCs can be matched with an available and node-compatible PV.
+----
+{"name" : "CheckVolumeBinding"}
 ----
 
-**_NoDiskConflict_** checks if the volume requested by a pod is available.
+The `CheckVolumeBinding` predicate must be enabled in non-default schedulers.
```

Comment:
I did not understand this statement? Can you elaborate?

Reply:
@ravisantoshgudimetla I cannot find the source for this statement. I think that means, if the user wants to use a scheduler other than the default, `CheckVolumeBinding` must be included in the scheduler policy.
@sunilcio Any thoughts on this?

Reply:
@ravisantoshgudimetla @mburke5678 I am also not sure about this statement. From the docs [1], I see `CheckVolumeBinding` evaluates if a pod can fit based on the volumes and this is not in default predicates.
[1] https://docs.openshift.com/container-platform/4.1/nodes/scheduling/nodes-scheduler-default.html#other-predicates_nodes-scheduler-default

Reply:
@ravisantoshgudimetla I guess you were checking about `CheckVolumeBinding` predicate due to my example `policy.cfg` config from my previous comment. If that is the case, then please ignore that as I was just trying to add non-default predicates to custom scheduler policy for testing. I have checked with below simple config for custom scheduler but after creating it, new pods are still using default scheduler.
```
{
"kind" : "Policy",
"apiVersion" : "v1",
"predicates" : [
        {"name" : "PodFitsHostPorts"},
        {"name" : "PodFitsResources"},
        {"name" : "NoDiskConflict"},
        {"name" : "NoVolumeZoneConflict"},
        {"name" : "MatchNodeSelector"},
        {"name" : "HostName"}
        ],
"priorities" : [
        {"name" : "LeastRequestedPriority", "weight" : 1},
        {"name" : "BalancedResourceAllocation", "weight" : 1},
        {"name" : "ServiceSpreadingPriority", "weight" : 1},
        {"name" : "EqualPriority", "weight" : 1}
        ]
}
```

Reply:
@sunilcio @ravisantoshgudimetla This might be where this statement came from:
https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.9.md#storage-1

I have commented out the line.